### PR TITLE
Add support for isInf, isNan, and iszero

### LINKED ISSUE #2750: af_isinf, af_isnan, af_iszero for f16 arrays convert inputs to f32
https://github.com/arrayfire/arrayfire/blob/c1e141691187c4a56f17fffbc52b832d3fa3d2c9/src/api/c/unary.cpp#L671-L674

### PR DESCRIPTION
**Pull request by umar456**
Fixes #2750

### CODE REVIEW

File: src/api/c/unary.cpp [hunk 1]
```diff
@@ -673,6 +673,7 @@ static af_err af_check(af_array *out, const af_array in) {
         switch (type) {
```
Comment (gaika) [inline src/api/c/unary.cpp]: `type` is  f32 when the input is f16.

Comment (gaika) [reply src/api/c/unary.cpp]: need `        if (in_type == f16) { type = f16; }` before that line

Comment (umar456) [reply src/api/c/unary.cpp]: ahh good catch.

File: src/api/c/unary.cpp [hunk 2]
```diff
@@ -669,10 +669,12 @@ static af_err af_check(af_array *out, const af_array in) {
 
         // Convert all inputs to floats / doubles / complex
         af_dtype type = implicit(in_type, f32);
+        if(in_type == f16) type = f16;
```
Comment (9prady9) [inline src/api/c/unary.cpp]: In this case, you can just call `af_dtype type = implicit(in_type, f16);`

Comment (umar456) [reply src/api/c/unary.cpp]: The implicit function will be used to convert integer types to floating point types. If we use f16 on the rhs, the integer types will be converted to f16 which has a pretty limited range compared to float.

Comment (9prady9) [reply src/api/c/unary.cpp]: I have checked that, Integer types are not supported in this function

Comment (9prady9) [reply src/api/c/unary.cpp]: We will probably never incude integer types for these three functions. Hence my comment

**Pull request merged**